### PR TITLE
Delete unused methods from CoreCLR PAL

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
https://github.com/dotnet/runtime/pull/91524#discussion_r1314767022

File: src/coreclr/pal/src/misc/sysinfo.cpp [hunk 1]
```diff
@@ -249,426 +237,3 @@ GetSystemInfo(
...
-    lpBuffer->ullAvailVirtual = lpBuffer->ullAvailPhys;
-
-    LOGEXIT("GlobalMemoryStatusEx returns %d\n", fRetVal);
-    PERF_EXIT(GlobalMemoryStatusEx);
-
-    return fRetVal;
-}
-
-bool
-PAL_ReadMemoryValueFromFile(const char* filename, uint64_t* val)
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/pal/src/misc/sysinfo.cpp:493]: I think this is used in a few other places.

Comment (am11) [reply src/coreclr/pal/src/misc/sysinfo.cpp:493]: It is used in cgroup, which also is now-unused code in PAL (GC has a separate copy). I think the overarching idea was to move all memory related PAL code to GC, which happened a couple of months ago, but these leftovers weren't cleaned up for some reason at the same time.

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/coreclr/pal/src/misc/sysinfo.cpp (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/pal/src/misc/sysinfo.cpp:493]: > cgroup is used to read processor quota limits outside the GC.

Confirmed. I just tried to remove it also and hit that :(

Comment (am11) [reply src/coreclr/pal/src/misc/sysinfo.cpp:493]: Ah, I see. Maybe in the future, we could rewire it so those values are read from single place (GC PAL).

**Pull request merged**